tools/specs/lib/combineTilesetSpec.js

```diff
@@ -0,0 +1,186 @@
+'use strict';
+var fsExtra = require('fs-extra');
+var path = require('path');
+var Promise = require('bluebird');
+var combineTileset = require('../../lib/combineTileset');
+var getFilesInDirectory = require('../../lib/getFilesInDirectory');
+var gzipTileset = require('../../lib/gzipTileset');
+var isGzippedFile = require('../../lib/isGzippedFile');
+
+var tilesetDirectory = './specs/data/TilesetOfTilesets/';
+var combinedDirectory = './specs/data/TilesetOfTilesets-combined';
+var combinedJson = './specs/data/TilesetOfTilesets-combined/tileset.json';
+var gzippedDirectory = './specs/data/TilesetOfTilesets-gzipped';
+
+function isJson(file) {
+    return path.extname(file) === '.json';
+}
+
+function getContentUrls(string) {
+    var regex = new RegExp('"url":"([^"]*)"', 'g');
+    var matches = [];
+    var match = regex.exec(string);
+    while (match !== null) {
+        matches.push(match[1]);
+        match = regex.exec(string);
+    }
+    return matches;
+}
+
+function getNumberOfTilesets(directory) {
+    return getFilesInDirectory(directory)
+        .then(function (files) {
+            var length = files.length;
+            var numberOfJsonFiles = 0;
+            for (var i = 0; i < length; ++i) {
+                if (isJson(files[i])) {
+                    ++numberOfJsonFiles;
+                }
+            }
+            return numberOfJsonFiles;
+        });
+}
+
+describe('combineTileset', function() {
+    afterEach(function(done) {
+        Promise.all([
+            fsExtra.remove(gzippedDirectory),
+            fsExtra.remove(combinedDirectory)
+        ]).then(function() {
+            done();
+        });
+    });
+
+    it('combines external tilesets into a single tileset', function (done) {
+        var combineOptions = {
+            inputDirectory : tilesetDirectory,
+            outputDirectory : combinedDirectory
+        };
+        expect(combineTileset(combineOptions)
+            .then(function() {
+                return getNumberOfTilesets(combinedDirectory);
+            })
+            .then(function(numberOfTilesets) {
+                expect(numberOfTilesets).toBe(1);
+                return fsExtra.readFile(combinedJson, 'utf8');
+            })
+            .then(function(contents) {
+                var matches = getContentUrls(contents);
+                expect(matches).toEqual(['parent.b3dm', 'tileset3/ll.b3dm', 'lr.b3dm', 'ur.b3dm', 'ul.b3dm']);
```

Comment:
What happens if there is an external tileset and more than one tileset.json refer to a single b3dm file? Will the copy function copy it to multiple directories?

Also, by extension, how will i3dm and the external glTF files for those be handled?